core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala

```diff
@@ -104,7 +104,7 @@ private[spark] class TypedConfigBuilder[T](
   /** Checks if the user-provided value for the config matches the validator. */
   def checkValue(validator: T => Boolean, errorMsg: String): TypedConfigBuilder[T] = {
     transform { v =>
-      if (!validator(v)) throw new IllegalArgumentException(errorMsg)
+      if (!validator(v)) throw new IllegalArgumentException(s"'$v' is invalid because: $errorMsg")
```

Comment:
```suggestion
      if (!validator(v)) throw new IllegalArgumentException(s"'$v' in ${parent.key} is invalid. $errorMsg")
```

Reply:
what about this? to make it sound similar with `checkValues`

Reply:
this is better. updated. thanks @HyukjinKwon